pycrires/pipeline.py

```diff
@@ -3726,108 +3726,90 @@ def correct_wavelengths(self, nod_ab: str = "A") -> None:
+            print(f"Reading spectra of nod {nod_ab}...", end="", flush=True)
 
-            # Get all spectral orders
-            spec_orders = np.sort([i[:5] for i in data.dtype.names if "WL" in i])
+            hdu_list = fits.open(fits_file)
 
-            print(f"\nCorrecting wavelength solution of detector {i_det+1}:")
-
-            # Loop over spectral orders
-            for spec_name in spec_orders:
-                # Extract WL, SPEC, and ERR for given order/detector
-                wavel = hdu_list[f"CHIP{i_det+1}.INT1"].data[spec_name + "_WL"]
-                spec = hdu_list[f"CHIP{i_det+1}.INT1"].data[spec_name + "_SPEC"]
-                err = hdu_list[f"CHIP{i_det+1}.INT1"].data[spec_name + "_ERR"]
+            print(" [DONE]")
 
-                spec = np.nan_to_num(spec)
-                err = np.nan_to_num(err)
 
-                spec_data = np.column_stack([wavel, spec, err])
+            for i_det in range(3):
+                # Get detector data
+                data = hdu_list[f"CHIP{i_det+1}.INT1"].data
 
-                # spec_filt = signal.medfilt(spec, kernel_size=201)
-                # poly_coef = np.polyfit(wavel, spec_filt, 3)
-                # poly_spec = np.poly1d(poly_coef)
-                # spec_data[:, 1] /= poly_spec(wavel)
+                # Get all spectral orders
+                spec_orders = np.sort([i[:5] for i in data.dtype.names if "WL" in i])
 
-                opt_result = optimize.minimize(
-                    corr_func,
-                    x0=(0.01, 1.0),
-                    args=(spec_data, transm_interp),
-                    method="Nelder-Mead",
-                    bounds=((-1.0, 1.0), (1.0, 1.0)),
-                    options={"maxiter": 1000, "xatol": 1e-4, "fatol": np.inf},
-                )
+                print(f"\nCorrecting wavelength solution of detector {i_det+1}:")
 
-                print(
-                    f"   - {spec_name} -> lambda = {opt_result.x[0]:.4f} "
-                    f"+ {opt_result.x[1]:.1f} * lambda'"
-                )
-
-                if not opt_result.success:
-                    warnings.warn(
-                        f"Could not converge with correcting the "
-                        f"wavelength solution of {spec_name[:-3]}."
+                # Loop over spectral orders
+                for spec_name in spec_orders:
+                    # Extract WL, SPEC, and ERR for given order/detector
+                    wavel = hdu_list[f"CHIP{i_det+1}.INT1"].data[spec_name + "_WL"]
+                    spec = hdu_list[f"CHIP{i_det+1}.INT1"].data[spec_name + "_SPEC"]
+                    err = hdu_list[f"CHIP{i_det+1}.INT1"].data[spec_name + "_ERR"]
+
+                    #Remove continuum and nans of spectra
+                    nans = np.isnan(spec)
+                    continuum = ndimage.gaussian_filter1d(spec[~nans], 100, mode='reflect')
+                    spec_flat = spec[~nans]/continuum - 1
```

Comment:
Why is the -1 needed? Maybe change to -1. such that it is clear that `spec_flat` is a float (although it works fine like this).